controller/src/main/java/org/jboss/as/controller/access/constraint/ApplicationTypeConstraint.java

```diff
@@ -143,5 +149,10 @@ protected int internalCompare(AbstractConstraintFactory other) {
             // We have no preference
             return 0;
         }
+
+        @Override
+        public Constraint getRequiredConstraint(Action.ActionEffect actionEffect, JmxAction action, JmxTarget target) {
+            return (isApplicationType(action.composeAction()) || isApplicationType(target)) ? APPLICATION : NON_APPLICATION;
```

Comment:
This can simply return NON_APPLICATION and isApplicationType() can be removed. The methods will always return false.